hazelcast/src/main/java/com/hazelcast/transaction/impl/xa/XAResourceImpl.java

```diff
@@ -107,7 +107,7 @@ public void start(Xid xid, int flags) throws XAException {
 
     private TransactionContext createTransactionContext(Xid xid) {
         XAService xaService = getService();
-        TransactionContext context = xaService.newXATransactionContext(xid, timeoutInSeconds.get(), false);
+        TransactionContext context = xaService.newXATransactionContext(xid, null, timeoutInSeconds.get(), false);
```

Comment:
if passed null, transcation context picks local member's uuid. The issue here is we were passing null for client too